Assets/Scripts/Beast/BeastBehavior.cs

```diff
@@ -4,21 +4,55 @@
 
 public class BeastBehavior : MonoBehaviour
 {
-
     public GameObject player;
+    public GameObject target;
+    public float fightDistance = 1.7f;
+
+    Animator animator;
 
     void Start()
     {
-
+        animator = GetComponent<Animator>();
     }
 
     void Update()
     {
-        Vector3 targetPosition = player.transform.position;
+        GameObject toInteract = (target == null ? player : target);
+
+        Vector3 targetPosition = toInteract.transform.position;
         Vector3 direction = (targetPosition - transform.position).normalized;
 
         transform.rotation = Quaternion.FromToRotation(Vector3.right, direction);
 
         transform.Translate(Vector3.right * Time.deltaTime);
```

Comment:
We'll probably want to extract this out to its own function later, but we can do that later when we handle mouse movement